### PR TITLE
KAFKA-15742: KRaft support in GroupCoordinatorIntegrationTest

### PR DESCRIPTION
**Pull request by wernerdv**
updated GroupCoordinatorIntegrationTest to support KRaft

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by wernerdv** [comment]
Hello, @mimaison @jolshan
Please, take a look.

https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-15086/lastCompletedBuild/testReport/kafka.api/GroupCoordinatorIntegrationTest/

**Comment by wernerdv** [comment]
@divijvaidya If you have a chance, could you take a look at this PR?

**Comment by jolshan** [comment]
`testGroupCoordinatorPropagatesOffsetsTopicCompressionCodec(String).quorum=kraft PASSED`

Looks like it is passing on all builds.

🥳

File: core/src/test/scala/integration/kafka/api/GroupCoordinatorIntegrationTest.scala [hunk 1]
```diff
@@ -33,18 +34,21 @@ class GroupCoordinatorIntegrationTest extends KafkaServerTestHarness {
...
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testGroupCoordinatorPropagatesOffsetsTopicCompressionCodec(quorum: String): Unit = {
     val consumer = TestUtils.createConsumer(bootstrapServers())
     val offsetMap = Map(
       new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, 0) -> new OffsetAndMetadata(10, "")
     ).asJava
     consumer.commitSync(offsetMap)
-    val logManager = servers.head.getLogManager
+    val logManager =
+      if (isKRaftTest()) brokers.head.logManager
```
Comment (jolshan) [inline core/src/test/scala/integration/kafka/api/GroupCoordinatorIntegrationTest.scala]: Can we use brokers.head.logManager for Zk tests too?

Comment (wernerdv) [reply core/src/test/scala/integration/kafka/api/GroupCoordinatorIntegrationTest.scala]: yep, fixed

**Review by jolshan — APPROVED** [review]
LGTM

**Pull request merged**